src/Bounds.cpp

```diff
@@ -1039,13 +1053,25 @@ class Bounds : public IRVisitor {
                     } else if (op->is_intrinsic(Call::bitwise_and) &&
                                a_interval.has_upper_bound() &&
                                b_interval.has_upper_bound()) {
-                        if (t.is_int()) {
-                            // Smaller than the larger of the two args
-                            interval.max = max(a_interval.max, b_interval.max);
-                        } else if (t.is_uint()) {
-                            // Smaller than both args
+                        bool a_positive = can_prove(a_interval.min >= 0);
+                        bool b_positive = can_prove(b_interval.min >= 0);
+                        if (t.is_uint() || (a_positive && b_positive)) {
+                            // Positive and smaller than both args
                             interval.max = min(a_interval.max, b_interval.max);
+                            interval.min = make_zero(t);
+                        } else if (t.is_int()) {
+                            if (b_positive) {
+                                interval.min = make_zero(t);
+                                interval.max = b_interval.max;
+                            } else if (a_positive) {
+                                interval.min = make_zero(t);
+                                interval.max = a_interval.max;
+                            } else {
+                                // Smaller than the larger of the two args
```

Comment:
So in this case, the min remains unbounded?

Reply:
Yeah, if neither arg is known to be positive then you can't say anything about the sign bit. All the others bits could cancel out and leave you with 0x80000000